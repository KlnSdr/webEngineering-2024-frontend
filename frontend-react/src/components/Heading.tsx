import React from "react";

interface HeadingProps {
  headingText: string;
}

const Heading: React.FC<HeadingProps> = ({ headingText }) => {
  return <h1 className="headline-with-lines">{headingText}</h1>;
};

export default Heading;
