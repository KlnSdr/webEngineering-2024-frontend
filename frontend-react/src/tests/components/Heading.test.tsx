import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Heading from "../../components/Heading"; // Adjust the import if the path is different

describe("Heading Component", () => {
    test("renders with correct heading text", () => {
        const headingText = "Test Heading";

        render(<Heading headingText={headingText} />);

        const headingElement = screen.getByRole("heading", { level: 1 });
        expect(headingElement).toHaveTextContent(headingText);
        expect(headingElement).toHaveClass("headline-with-lines");
    });
});
