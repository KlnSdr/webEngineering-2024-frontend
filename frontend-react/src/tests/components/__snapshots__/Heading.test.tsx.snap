// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component renders with correct heading text 1`] = `
<DocumentFragment>
  <h1
    class="headline-with-lines"
  >
    Test Heading
  </h1>
</DocumentFragment>
`;
